mailpoet/assets/js/src/segments/dynamic/dynamic.tsx

```diff
@@ -8,10 +8,13 @@ import { Editor } from 'segments/dynamic/editor';
 import { DynamicSegmentList } from 'segments/dynamic/list';
 import { SegmentTemplates } from 'segments/dynamic/templates';
 import * as ROUTES from 'segments/routes';
+import { createStore } from 'segments/dynamic/store';
 
 const container = document.getElementById('dynamic_segments_container');
 
 function App(): JSX.Element {
+  createStore();
```

Comment:
This place is actually a risky place to put it (and so was the original one), because it's in the rendering body of a component. The function can be called multiple times, if multiple re-renders occur for any reason.

In most scenarios, the solution is to put the code in `useEffect(..., [])`, but for a store, I think the best solution is to instantiate it before creating the React app. I mean at the end of this file, something like this:

```ts
if (container) {
  registerTranslations();
  createStore(); // added here
  ReactDOM.render(<App />, container);
}
```

Reply:
Good point! I added a new commit moving `createStore()` to the place you suggested.